public/js/controllers.js

```diff
@@ -43,6 +43,18 @@ angular.module('starter.controllers', [])
         var refresh = $scope.dataset?false:true;
         $scope.getData(refresh);
     });
+
+    $scope.$on(ConstEvents.UPDATE_DISTRICTS, function(evnt, newSelectedDistricts) {
```

Comment:
UPDATE_DISTRICTS event is fired after users update the districts filter. I tried not to use $rootScope to store these settings, because rootScope is like global namespace, instead handled the changes with events. Any other view that wants is to be effected by this change can listen to this event.